### DISCUSSION TITLE
Automatic Update all LXC Container with Script possible?

### DISCUSSION THREAD

**Discussion by mapime** [tteck/Proxmox#2445] [General]
Hello,
is it possible to run that helper scripts on the LXC Container automatically for unattended updates?
Like Nginx Proxy Manager LXC with that command: 
"bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/nginxproxymanager.sh)""

I found the LXC Updater, but I think its only for Update the System OS like Debian with apt update - right?

The alternative is to install docker with Watchtower...

Maybe you can help me. Thanks!

**Comment by tteck** [comment]
While anything may be possible, I choose not to develop such a script. Unattended application updates are ill-advised.

**Reply by mapime** [reply]
I knew the the trouble with unattended updates. But in this case it would be a significant relief. I have several containers and the update process always eats up a lot of time.

It is better to have a weekly backup and then the option for automatic updates.
Restoring an LXC is very easy.

In addition, automatic updates for LXC which are available on the Internet are always better or not?

Thanks for your reply

**Reply by bobloadmire** [reply]
With adequate backups, I don't really see why it would be an issue, if it works 99% of the time, which it probably would, it would still be a huge time saver, even if you had to restore an LXC once a year.

**Comment by arteck** [comment]
i use this 

```

#!/bin/bash
# update all containers

apt autoremove

# list of container ids we need to iterate through
containers=$(pct list | tail -n +2 | cut -f1 -d' ')

function update_container() {
  container=$1
  echo "------------------->        [Info] Updating $container"
  # to chain commands within one exec we will need to wrap them in bash
  pct exec $container -- bash -c "apt update && apt upgrade -y && apt autoremove -y"
}

for container in $containers
do
  status=`pct status $container`
  if [ "$status" == "status: running" ]; then
#    echo [Info] updating $container
    update_container $container
  fi
done; wait
```

**Reply by mapime** [reply]
Thanks for your reply.
This Script is updating the container with bash -c "apt update && apt upgrade -y && apt autoremove -y" ,right?
But does this command also update my nginx on the container, or should I substitute the bash command? 
It does not work on other containeres

Sorry for the question:
where do you save this script? here /bin/bash? I dont think.
And how do you run it regularly?

Thanks so much!
